### PR TITLE
chore(docs): update get-doc

### PR DESCRIPTION
**Pull request by icfantv**
* update doc gen script to iterate over components and generate the component
  documentation JSON to the appropriate component's demo directory.

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
As commented on slack, I don't think it is the best approach - we should rather generate JSON files in memory, without creating temp files. Also I would scan fs for components instead of hard-coding a list of them.

I can help / we can pair on wiring doc generation with this script.

**Comment by masaanli** [comment]
Do you guys have an public slack channel?

**Comment by pkozlowski-opensource** [comment]
> Do you guys have an public slack channel?

Nope, sorry, not at the moment :-/ But you can always reach out to use here on GH

**Comment by icfantv** [comment]
@pkozlowski-opensource, I went the scan route originally, but we have that `utils` directory in there along with our components.  If we intend for those to be public and supported then scanning is fine.  Thoughts?

**Comment by pkozlowski-opensource** [comment]
> I went the scan route originally, but we have that utils directory in there along with our components

Exclude `utils` for now. Still I think it is better to have one special case (exclude `utlis`) as compared to listing all directives by hand.

**Comment by icfantv** [comment]
Agreed.  It certainly scales better.

**Comment by icfantv** [comment]
@pkozlowski-opensource updated to be in memory and to filter out utils.  right now the `get-doc.js` script holds the component map - is that ok or do we want to pull it out and store it elsewhere?

File: misc/get-doc.js [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
+  return result;
+}
+
+const componentMap = new Map();
+const items = fs.readdirSync('./src');
+items.forEach(component => {
+  if (!excluded.has(component) && isDirectory(`./src/${component}`)) {
+    const names = getFileNames(component);
+    if (names.every(name => fileExists(name))) {
+      componentMap.set(component, JSON.stringify(doc(names)));
```
Comment (pkozlowski-opensource) [inline misc/get-doc.js]: Why are you calling `JSON.stringify` here? Isn't it going to make your life harder when generating API docs?

**Comment by pkozlowski-opensource** [comment]
>  right now the get-doc.js script holds the component map - is that ok or do we want to pull it out and store it elsewhere?

Personally I wouldn't store it anywhere - would just have a function call that returns data. But it is your call.

File: misc/get-doc.js (hunk 1 above)
Comment (icfantv) [inline misc/get-doc.js]: wasn't thinking.  will remove.

**Comment by icfantv** [comment]
noted.

**Comment by icfantv** [comment]
updated.

File: misc/get-doc.js [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+    result = true;
+  }
+  catch (e) {
+    console.error(`the specified file does not exist: ${file}`)
+  }
+
+  return result;
+}
+
+function getComponentMap() {
```
Comment (pkozlowski-opensource) [inline misc/get-doc.js]: @icfantv I think that we should simply use an existing globbing library, ex.: https://www.npmjs.com/package/glob

**Comment by pkozlowski-opensource** [comment]
@icfantv I think that we can have it reduced to 2-liner by using existing npm libraries. See an alternative implementation in #236

**Pull request closed without merging**